spaceMissionManager-service/src/test/java/cz/muni/fi/services/CraftComponentServiceTest.java

```diff
@@ -0,0 +1,263 @@
+            mockedCraftComponent.setId(id);
+            craftComponents.put(id, mockedCraftComponent);
+            return mockedCraftComponent;
+        });
+
+        when(craftComponentDao.updateComponent(any(CraftComponent.class))).then(invoke -> {
+            CraftComponent mockedCraftComponent = invoke.getArgumentAt(0, CraftComponent.class);
+
+            if (mockedCraftComponent == null) {
+                throw new IllegalArgumentException("CraftComponent is null");
+            }
+            if (mockedCraftComponent.getId() == null) {
+                throw new IllegalArgumentException("CraftComponent is not in DB");
+            }
+            if (mockedCraftComponent.getName() == null) {
+                throw new IllegalArgumentException("CraftComponents name cant be null");
+            }
+            if (checkCraftComponentsNameDuplicity(mockedCraftComponent.getName(), mockedCraftComponent.getId())) {
+                throw new IllegalArgumentException("CraftComponents name already exist");
+            }
+
+            craftComponents.replace(mockedCraftComponent.getId(), mockedCraftComponent);
+            return mockedCraftComponent;
+        });
+
+        when(craftComponentDao.removeComponent(any(CraftComponent.class))).then(invoke -> {
+            CraftComponent mockedCraftComponent = invoke.getArgumentAt(0, CraftComponent.class);
+            if (mockedCraftComponent == null) {
+                throw new IllegalArgumentException("CraftComponent is null");
+            }
+            if (mockedCraftComponent.getId() == null) {
+                throw new IllegalArgumentException("Craftcomponent is not in DB");
+            }
+            craftComponents.remove(mockedCraftComponent.getId(), mockedCraftComponent);
+            return mockedCraftComponent;
+        });
+
+        when(craftComponentDao.findComponentById(anyLong())).then(invoke -> {
+            Long id = invoke.getArgumentAt(0, Long.class);
+            if (id == null) {
+                throw new IllegalArgumentException("id is null");
+            }
+            return craftComponents.get(id);
+        });
+
+        when(craftComponentDao.findAllComponents())
+                .then(invoke -> Collections.unmodifiableList(new ArrayList<>(craftComponents.values())));
+    }
+
+    @BeforeMethod
+    public void beforeTest() {
+        craftComponents.clear();
+        craftComponent1 = TestUtils.createCraftComponent("craftComponent1");
+        craftComponent2 = TestUtils.createCraftComponent("craftComponent2");
+        craftComponent3 = TestUtils.createCraftComponent("craftComponent3");
+
+        craftComponent1.setId(1L);
+        craftComponent2.setId(2L);
+        craftComponent3.setId(3L);
+
+        craftComponents.put(1L, craftComponent1);
+        craftComponents.put(2L, craftComponent2);
+        craftComponents.put(3L, craftComponent3);
+    }
+
+    @Test
+    public void createCraftComponent() throws DataAccessException {
```

Comment:
Should test throw an exception? Why?

Reply:
When using a method which can throw unchecked exception you have to add it to function definition